mscore/pagesettings.cpp

```diff
@@ -270,311 +321,428 @@ void PageSettings::orientationClicked()
-      preview->score()->style().set(Sid::pagePrintableWidth, val - (oddPageLeftMargin->value() + evenPageLeftMargin->value()) * f);
-      updatePreview(0);
+      Score*       prev  = preview->score();
+      QPageSize&   psize = prev->style().pageSize();
+      MPageLayout& odd   = prev->style().pageOdd();
+      MPageLayout& even  = prev->style().pageEven();
+
+      // Has anything changed? This prevents extra undo commands on the stack
+      // when the user clicks Apply or OK with no changes.
+      if (score->styleB(Sid::pageTwosided) == twosided->isChecked()
+       && score->pageNumberOffset()        == prev->pageNumberOffset()
+       && score->spatium()                 == prev->spatium()
+       && score->style().pageSize().id()   == prev->style().pageSize().id())
+            {
+            MPageLayout& sOdd  = score->style().pageOdd();
+            MPageLayout& sEven = score->style().pageEven();
+            if (sOdd.units()               == odd.units()
+             && sOdd.orientation()         == odd.orientation()
+             && sOdd.widthPoints()         == odd.widthPoints()
+             && sOdd.heightPoints()        == odd.heightPoints()
+             && sOdd.leftMarginPoints()    == odd.leftMarginPoints()
+             && sOdd.rightMarginPoints()   == odd.rightMarginPoints()
+             && sOdd.topMarginPoints()     == odd.topMarginPoints()
+             && sOdd.bottomMarginPoints()  == odd.bottomMarginPoints()
+             && sEven.topMarginPoints()    == even.topMarginPoints()
+             && sEven.bottomMarginPoints() == even.bottomMarginPoints())
+                  {
+                  return; // nothing has changed
+                  }
+            }
+      if (runCmd)
+            score->startCmd();
+
+      score->undoChangePageNumberOffset(pageOffset->value() - 1); // why isn't this a style??
+      score->undoChangeStylePtrs(psize, odd, even);
+      score->undoChangeStyleVal(Sid::spatium,          prev->spatium());
+      score->undoChangeStyleVal(Sid::pageTwosided,     twosided->isChecked());
+      score->undoChangeStyleVal(Sid::pageSize,         int(psize.id()));
+      score->undoChangeStyleVal(Sid::pageUnits,        pageUnits [int(odd.units())].key());
+      score->undoChangeStyleVal(Sid::pageOrientation,  pageOrient[int(odd.orientation())]);
+      score->undoChangeStyleVal(Sid::pageFullWidth,    odd.widthPoints());
+      score->undoChangeStyleVal(Sid::pageFullHeight,   odd.heightPoints());
+      score->undoChangeStyleVal(Sid::marginOddLeft,    odd.leftMarginPoints());
+      score->undoChangeStyleVal(Sid::marginOddRight,   odd.rightMarginPoints());
+      score->undoChangeStyleVal(Sid::marginOddTop,     odd.topMarginPoints());
+      score->undoChangeStyleVal(Sid::marginOddBottom,  odd.bottomMarginPoints());
+      score->undoChangeStyleVal(Sid::marginEvenTop,    even.topMarginPoints());
+      score->undoChangeStyleVal(Sid::marginEvenBottom, even.bottomMarginPoints());
+
+      // older 3.01 styles
```

Comment:
Older? There is one 3.01 file format. A developer won't understand this comment.